codegen/templates/tchannel_client.tmpl

```diff
@@ -70,22 +74,37 @@ func {{$exportName}}(deps *module.Dependencies) Client {
 	{{end -}}
 	sc.Peers().Add(ip + ":" + strconv.Itoa(int(port)))
 
-	var scAltName string
-	if deps.Default.Config.ContainsKey("clients.{{$clientID}}.staging.serviceName") {
-		scAltName = deps.Default.Config.MustGetString("clients.{{$clientID}}.staging.serviceName")
-		ipAlt := deps.Default.Config.MustGetString("clients.{{$clientID}}.staging.ip")
-		portAlt := deps.Default.Config.MustGetInt("clients.{{$clientID}}.staging.port")
-
-		scAlt := deps.Default.Channel.GetSubChannel(scAltName, tchannel.Isolated)
-		scAlt.Peers().Add(ipAlt + ":" + strconv.Itoa(int(portAlt)))
-	} else if deps.Default.Config.ContainsKey("clients.staging.all.serviceName") {
-		scAltName = deps.Default.Config.MustGetString("clients.staging.all.serviceName")
-		ipAlt := deps.Default.Config.MustGetString("clients.staging.all.ip")
-		portAlt := deps.Default.Config.MustGetInt("clients.staging.all.port")
-
-		scAlt := deps.Default.Channel.GetSubChannel(scAltName, tchannel.Isolated)
-		scAlt.Peers().Add(ipAlt + ":" + strconv.Itoa(int(portAlt)))
-	}
+	/*Ex:
+	{
+	  "clients.rider-presentation.alternates": {
+		"routingConfigs": [
+		  {
+			"headerName": "x-api-environment",
+			"headerValue": "*",
+			"serviceName": "presentation-sandbox"
+		  },
+		  {
+			"headerName": "RTAPI-Container",
+			"headerValue": "test*",
+			"serviceName": "mpx-prism"
+		  }
+		],
+		"servicesDetail": {
+		  "presentation-sandbox": {
+			"ip": "127.0.0.1",
+			"port": 5437
+		  },
+		  "mpx-prism": {
+			"ip": "127.0.0.1",
+			"port": 12958
+		  }
+		}
+	  }
+	}*/
+	var re ruleengine.RuleEngine
```

Comment:
`ruleengine` is a bit generic and not descriptive of how it's used. You might rename it as `peerchooser` or `peerselector`?